src/components/icon/examples/icon-image.tsx

```diff
@@ -0,0 +1,148 @@
+import { Component, h } from '@stencil/core';
+
+/**
+ * Displaying an image instead of an icon
+ *
+ * Because the `limel-icon` component is used by several other components
+ * already, we have added the ability to display an image instead of an icon.
+ * This makes it possible to display a user avatar, for example, anywhere that
+ * an icon is used.
+ *
+ * To render an image, simply provide the URL to the image as the `name`
+ * property.
```

Comment:
this explanation is great for the PR or for this commit, but feels a bit weird for the consumers in the docs I'd say.
Maybe better to say something along these lines:

> If you can't find a proper icon, feel free to add the path to your icon, instead the icon name. You can do the same in any other component that uses the `Icon` interface.


Also, I assume that in this case, the `color` and `backgroundColor` properties won't work anymore, even if the path was pointing to an SVG file, since we're rendering an `<img ` tag. So maybe worth to mention it somewhere, or make sure that svgs are treated as inlined svgs.

Reply:
Yeah, I was a little conflicted about the description. I kind of wanted to defend adding this somewhat weird ability to `limel-icon`… 😅 Or, if not defend, then at least explain why it's there.

I've actually had the thought that we could check if the string provided ends in .svg, and if it does, inject it in the page the same way we do with our own icons. But before doing that, we'd have to check that doing so doesn't create any vulnerabilities that wouldn't be created anyway by letting people load external SVGs in an `img` tag. (If the latter also creates vulnerabilities, we might want to filter what types of images we allow.)

For now, I'm pretty sure I did mention somewhere that it's possible to give a URL to an SVG, but setting the colour through CSS won't work when you do that.

Reply:
Regarding treating external SVGs the same way we treat our own icons though, I realised we _absolutely cannot_ do that. If we did, an attacker could inject any content they wanted by simply adding it to a file ending in `.svg`, and we'd happily inject it for them! 😱

We could probably validate that the content is actually SVG and stuff, but that would make the whole thing a lot more complex. Being able to load external SVG icons was not in the scope for the task here, so I'm not going to focus on that.